server/api/impl/admin/synchronize.go

```diff
@@ -180,7 +113,121 @@ func (adm *Admin) UninstallApp(appID apps.AppID) error {
 	return nil
 }
 
-func (adm *Admin) RegisterApp(manifest *apps.Manifest) error {
+func getAppsForRegistration(path string) (apps.AppVersionMap, error) {
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, errors.Wrapf(err, "can't read %s file", path)
+	}
+	var apps *AppVersions
+	if err := json.Unmarshal(data, &apps); err != nil || apps == nil {
+		return nil, errors.Wrapf(err, "can't unmarshal %s file", appsJSONFile)
+	}
+
+	return apps.Apps, nil
+}
+
+func (adm *Admin) populateManifests(appVersions apps.AppVersionMap) {
```

Comment:
s/`populateManifests`/`getAndStoreManifests`? (maybe populate is not the right action here).